# CÂU 4: PHÂN TÍCH HIỆU QUẢ KÊNH TRUY CẬP WEBSITE

**1. Yêu cầu bài toán:**
Trong file `web_traffic.csv`, nguồn truy cập (`traffic_source`) nào có tỷ lệ thoát trung bình (`bounce_rate`) thấp nhất tính trên tất cả các ngày xuất hiện của nguồn đó?

**2. Phương pháp tiếp cận & Suy luận:**
Để xác định kênh truyền thông mang lại người dùng chất lượng nhất (tương tác sâu, ít thoát trang), chúng ta thực hiện:
* **Bước 1:** Tải dữ liệu từ file `web_traffic.csv`.
* **Bước 2:** Nhóm dữ liệu theo cột `traffic_source`.
* **Bước 3:** Tính giá trị trung bình (`mean`) của cột `bounce_rate` cho từng nguồn truy cập.
* **Bước 4:** Sắp xếp kết quả theo thứ tự tăng dần và xác định nguồn có giá trị nhỏ nhất.

**3. Lý do chọn phương pháp:**
* **Sử dụng `groupby` kết hợp `mean()`:** Tỷ lệ thoát biến động theo từng ngày. Việc tính giá trị trung bình trên toàn bộ dòng dữ liệu giúp loại bỏ các yếu tố nhiễu tức thời (như các ngày có khuyến mãi đột biến), từ đó phản ánh đúng bản chất "giữ chân người dùng" của kênh đó trong dài hạn.
* **Tính tối ưu:** Thao tác này xử lý trực tiếp trên cột `bounce_rate` của bảng `web_traffic`, đảm bảo tốc độ tính toán nhanh nhất mà không cần thực hiện các phép nối bảng phức tạp.


In [1]:
# Import thư viện
import pandas as pd
from google.colab import drive

# Kết nối với Google Drive
drive.mount('/content/drive')

Mounted at /content/drive


**Thực hiện Bước 1 & Bước 2: Tải dữ liệu và Nhóm theo nguồn truy cập**

Chúng ta nạp dữ liệu lưu lượng web để bắt đầu phân tích hành vi người dùng theo từng nguồn dẫn.

In [2]:
# 1. Khai báo đường dẫn
file_path = '/content/drive/MyDrive/datathon-2026-round-1/web_traffic.csv'

# 2. Đọc dữ liệu
df_traffic = pd.read_csv(file_path)

# Hiển thị cấu trúc dữ liệu để kiểm tra các cột bounce_rate và traffic_source
df_traffic[['date', 'traffic_source', 'bounce_rate']].head()

,date,traffic_source,bounce_rate
0,2013-01-01,organic_search,0.00514
1,2013-01-02,organic_search,0.00406
2,2013-01-03,direct,0.00401
3,2013-01-04,direct,0.00562
4,2013-01-05,referral,0.00525


**Thực hiện Bước 3 & Bước 4: Tính toán tỷ lệ thoát trung bình**

Ta thực hiện gom nhóm và tính toán để tìm ra kênh có tỷ lệ thoát thấp nhất (chỉ số càng thấp chứng tỏ kênh đó càng chất lượng).

In [4]:
# Gom nhóm và tính trung bình bounce_rate, sau đó sắp xếp tăng dần
traffic_performance = df_traffic.groupby('traffic_source')['bounce_rate'].mean().sort_values()

print("Tỷ lệ thoát trung bình theo từng nguồn truy cập:")
print(traffic_performance)

# Xác định nguồn có tỷ lệ thoát thấp nhất
best_source = traffic_performance.idxmin()
min_bounce = traffic_performance.min()

print(f"\n--> Nguồn truy cập có tỷ lệ thoát trung bình thấp nhất là: {best_source} ({min_bounce:.4f})")

Tỷ lệ thoát trung bình theo từng nguồn truy cập:
traffic_source
email_campaign    0.004458
social_media      0.004476
paid_search       0.004478
referral          0.004499
organic_search    0.004504
direct            0.004511
Name: bounce_rate, dtype: float64

--> Nguồn truy cập có tỷ lệ thoát trung bình thấp nhất là: email_campaign (0.0045)


**KẾT LUẬN CUỐI CÙNG:**
Dựa trên kết quả tính toán từ tập dữ liệu `web_traffic.csv`, nguồn truy cập có tỷ lệ thoát trung bình thấp nhất là **email_campaign** (với giá trị khoảng 0.0045).

Đối chiếu với các phương án của đề bài:

A) organic_search

B) paid_search

C) email_campaign

D) social_media

**=> Đáp án được chọn là: C**